src/LinAlg/SAM.C

```diff
@@ -208,6 +208,7 @@ bool SAM::initSystemEquations ()
   std::cout <<"SAM::initSystemEquations()"<< std::endl;
 #endif
   if (!msc && ndof > 0) return false;
+  if (!msc && nceq > 0) return false;
   if (!mpmceq && nceq > 0) return false;
```

Comment:
Its then better to combine L211 and L212 into `if ((!msc || !mpmceq) && nceq > 0) return false;`.

Reply:
problem is that's not equivalent. i also need to check ndof to satisfy the sanitizer.
```
if (!msc && (ndof > 0 || nceq > 0)) return false;
if (!mpmceq && nceq > 0) return false;
```
does the job, the suggestion does not.

Reply:
my bad, i misread. that does indeed do the job.